pkg/common/cluster/healthchecks/clusterversionoperator.go

```diff
@@ -33,7 +33,28 @@ func CheckCVOReadiness(configClient configclient.ConfigV1Interface, logger *log.
 	var metadataState []string
 
 	for _, v := range cvInfo.Status.Conditions {
-		if (v.Type != "Available" && v.Status != "False") && v.Type != "Upgradeable" && v.Type != "RetrievedUpdates" {
+		conditionError := false
+
+		// Available must be true
+		if v.Type == "Available" && v.Status != "True" {
+			conditionError = true
+		}
+
+		// Look for any CVO condition which is 'true' which is not a known 'acceptable' true condition
+		if v.Status == "True" {
+			foundAllowedCondition := false
+			for _, allowedCondition := range []v1.ClusterStatusConditionType{"Available", "Upgradeable", "RetrievedUpdates", "ReleaseAccepted"} {
+				if v.Type == allowedCondition {
+					foundAllowedCondition = true
+				}
+			}
+			if !foundAllowedCondition {
+				// A condition is true which is not an accepted-true condition
+				conditionError = true
+			}
+		}
+
+		if conditionError {
 			metadataState = append(metadataState, fmt.Sprintf("%v", v))
 			logger.Printf("CVO State not complete: %v: %v %v", v.Type, v.Status, v.Message)
 			success = false
```

Comment:
```suggestion
		switch v.Type {
		case "Available":
			// Available must be true
			if v.Status == "True" {
				continue
			}
		case "Upgradeable", "RetrievedUpdates", "ReleaseAccepted":
			// These conditions don't matter to readiness state
			continue
		default:
			// Ignore any condition that isn't true
			// Any other true condition will return an error
			if v.Status != "True" {
				continue
			}
		}

		metadataState = append(metadataState, fmt.Sprintf("%v", v))
		logger.Printf("CVO State not complete: %v: %v %v", v.Type, v.Status, v.Message)
		success = false
```

Reply:
Thanks, that switch-around reads a lot better. Applied this change.